Sources/Sentry/SentryTracer.m

```diff
@@ -9,12 +9,18 @@
 
 static const void *spanTimestampObserver = &spanTimestampObserver;
 
+@interface
+SentryTracer ()
+
+@property (nonatomic, strong) SentrySpan *rootSpan;
+@property (nonatomic, strong) NSMutableArray<id<SentrySpan>> *children;
+@property (nonatomic, strong) SentryHub *hub;
+@property (nonatomic) SentrySpanStatus finishStatus;
+@property (nonatomic) BOOL isWaitingChildren;
```

Comment:
Using private properties instead of private variables will help when creating tests, we can access this using Dynamic.
With this we don't need to make each test an integration test to check the result, so the test will end up simpler, easier to understand, self-contained, and testing just the target class.

Reply:
I also discovered that using private properties is handy when using Dynamic 👍